R (#498)
Codecademy/docs: content/r/concepts/data-types/data-types.md

---
Title: 'Data Types'
Description: 'R supports many data types, each with different uses and characteristics.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Data Types'
  - 'Integers'
  - 'Booleans'
  - 'Strings'
CatalogContent:
  - 'learn-r'
  - 'paths/computer-science'
---

R supports many data types, each with different uses and characteristics. Variables can hold different data types, and are not limited to storing the data type they were initially assigned.

```r
# This is legal in R

# Assign an integer data type
var1 <- 1L

# Assign a string data type
var1 <- "Hello world!"
```

R supports the following basic data types:

> - The `numeric` type for whole numbers or decimals. (i.e. 5.12, 16)

Reviewer: ```suggestion
- The `numeric` type for whole numbers or decimals (5.12, 16).
```
perhaps we can edit this to eliminate 'i.e.' on each line

---
Title: 'Data Types'

> Description: 'R has a number of different data types with different uses and characteristics.'

Reviewer 2: ```suggestion
Description: 'R supports many data types, each with different uses and characteristics.'
```

Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Data Types'
  - 'Integers'
  - 'Booleans'
  - 'Strings'
CatalogContent:
  - 'learn-r'
  - 'paths/computer-science'
---

> R has a number of different data types with different uses and characteristics.
> R variables can hold different data types, and are not limited to storing the data type they were initially assigned.

Reviewer 2: ```suggestion
R supports many data types, each with different uses and characteristics. Variables can hold different data types, and are not limited to storing the data type they were initially assigned.
```

```r
# This is legal in R

# Assign an integer data type
var1 <- 1L

# Assign a string data type
var1 <- "Hello world!"
```

R supports the following basic data types:

> - `numeric`: A whole or decimal number. (i.e. 5.12, 16)
> - `integer`: A whole number followed by `L`. (i.e. 100L)
> - `complex`: A number with an imaginary part. (i.e. 5 + 7i, `i` denoting the imaginary part)
> - `character`: A string of characters, (i.e. "Hello World!")
> - `logical`: A Boolean value. (i.e. `TRUE` or `FALSE`)

Reviewer 2: ```suggestion
- The `numeric` type for whole numbers or decimals. (i.e. 5.12, 16)
- The `integer` type for whole numbers followed by an `L`. (i.e. 100L)
- The `complex` type for numbers with an imaginary part, denoted by an `i`. (i.e. 5 + 7i)
- The `character` type for strings of characters. (i.e. "Hello World!")
- The `logical` for Boolean values. (i.e. `TRUE` or `FALSE`)
```
